### ISSUE TITLE
Update Python version

### ISSUE THREAD

**Issue by forsyth2** [E3SM-Project/zstash#220]
Update Python version in `.github/workflows/build_workflow.yml`

**Comment by forsyth2** [comment]
@tomvothecoder I notice in https://github.com/E3SM-Project/e3sm_diags/pull/621/files you have:
```
    strategy:
      matrix:
        python-version: ["3.8", "3.9", "3.10"]
```
However, `zstash` has multiple `python-version` specified in https://github.com/E3SM-Project/zstash/blob/main/.github/workflows/build_workflow.yml. Does setting this in `matrix` negate the need to have it anywhere else?

**Comment by tomvothecoder** [comment]
To break it down, the build workflow contains jobs such as `pre-commit-hooks` and `build`. These jobs might use some prebuilt `actions` such as `actions/setup-python@v2`, which includes a setting for `python-version`. 

Example: 
https://github.com/E3SM-Project/zstash/blob/30d1784f6de2ebe722170a654a83b4c380a45a55/.github/workflows/build_workflow.yml?rgh-link-date=2022-09-12T21%3A59%3A36Z#L11-L27

In the E3SM diags build workflow, we specify a `matrix`  for the `build` job because we want to rerun this job (and the test suite) for different versions of Python. We can do the same for other jobs, but we don't need to.
```
  build:
    name: Build (Python ${{ matrix.python-version }})
    runs-on: "ubuntu-latest"
    timeout-minutes: 20
    defaults:
      run:
        shell: bash -l {0}
    strategy:
      matrix:
        python-version: ["3.8", "3.9", "3.10"]
```

**Specifying a matrix is not needed for `zstash` unless you have a test suite you want to run across different versions of Python. Otherwise, you can simply update `python-version` from 3.7 to 3.10 wherever needed.**

**Comment by forsyth2** [comment]
Fixed by #224
